src/server.c

```diff
@@ -6121,15 +6121,47 @@ sds genRedisInfoString(const char *section) {
 }
 
 void infoCommand(client *c) {
-    char *section = c->argc == 2 ? c->argv[1]->ptr : "default";
+    if (c->argc == 1) {
 
-    if (c->argc > 2) {
-        addReplyErrorObject(c,shared.syntaxerr);
+        sds info = genRedisInfoString("default");
+        addReplyVerbatim(c,info,sdslen(info),"txt");
+        sdsfree(info);
+        return;
+    }
+
+    int defsections = 0, allsections = 0;
+    // first time find all/default flag
+    for (int i = 1; i < c->argc; i++) {
+
+        defsections = !strcasecmp(c->argv[i]->ptr,"default");
+        allsections = !strcasecmp(c->argv[i]->ptr,"all");
+    }
+
+    if (defsections || allsections) {
+
+        sds info = allsections ? genRedisInfoString("all") : genRedisInfoString("default");
+        addReplyVerbatim(c,info,sdslen(info),"txt");
+        sdsfree(info);
         return;
     }
```

Comment:
so if someone does `INFO default commandstats`, he'll get just default, and no commandstats?